pilot/pkg/model/xds_cache.go

```diff
@@ -222,36 +210,39 @@ func (l *lruCache) assertUnchanged(key string, existing *discovery.Resource, rep
 	}
 }
 
-func (l *lruCache) Add(entry XdsCacheEntry, token CacheToken, value *discovery.Resource) {
-	if !entry.Cacheable() {
+func (l *lruCache) Add(entry XdsCacheEntry, pushReq *PushRequest, value *discovery.Resource) {
+	if !entry.Cacheable() || pushReq == nil {
 		return
 	}
+	// Start unset
+	if pushReq.Start.Equal(time.Time{}) {
+		panic("token cannot be empty!")
+	}
+	// TODO: It will not overflow until year 2262
+	token := CacheToken(pushReq.Start.UnixNano())
```

Comment:
In the old token model, tokens were generated+read under a consistent lock. This guaranteed consistency in them.

In the new model, the token is generated by calling time.Now() across many goroutines. Does go guarantee that time.Now is monotonically increasing *across goroutines*?

If not, we may run into issues

Reply:
It is not monotonically increasing, the StartTime is set in a debouncing, single thread. In 
ProxyUpdate, it can also be set, i donot think it matters even if duplicate time used.